### PR TITLE
Resume upload

### PR DESCRIPTION
**Pull request by Yadu-M**
-added a sample ui component for resume upload
-uploads resume to firebase storage
-registers the resume by adding a document to firestore
-added function to retrieve resume-document reference

### CODE REVIEW

File: src/App.tsx
Comment (namoosir) [inline src/App.tsx:1]: work with @altaafj to integrate the router properly

File: src/firebase/database.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  description: string | null,
+  url: string,
+  uid: string,
+  jobId: string | null,
+  filename: string
+}
+
+export const addResume = async (data: resume) => {  
+  try {
+    const db = getFirestore();
```
Comment (namoosir) [inline src/firebase/database.ts]: this works without passing in the firebase app variable?

File: src/firebase/database.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    })
+  } catch (error) {
+    console.error(`Something went wrong while trying to add resume for user(${data.uid}): ${error}`)
+    throw new Error(`${error}`)
+  }
+}
+
+export const getUserResume = async (uid: string, filename: string) => {
+  try {
+    const db = getFirestore();
```
Comment (namoosir) [inline src/firebase/database.ts]: repetitive and reinitializing this multiple times, use the context in a component and then pass in from the component into this function

File: src/firebase/database.ts
Comment (namoosir) [inline src/firebase/database.ts:1]: we should not put all database related stuff into one file, split them up into different services like services/resumeService.ts which will do crud on resume, etc.

File: src/firebase/storage.ts
Comment (namoosir) [inline src/firebase/storage.ts:1]: use a linter, the indentation is messed up, use 2 spaces

File: src/components/ResumeForm.tsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  type changeEvent = React.ChangeEvent<HTMLInputElement> | undefined
+  type submitEvent = React.FormEvent<HTMLFormElement>
+
+  const [files, setFiles] = useState<Array<File>>()
+
+  function submitHandler(event: submitEvent) {
+    event.preventDefault();
+    if (files) {
+      for (const file of files) {
+        uploadResume(file, "uid", "jobid", "description")
```
Comment (namoosir) [inline src/components/ResumeForm.tsx]: is this incomplete?

**Review by namoosir — CHANGES_REQUESTED** [review]

File: src/components/ResumeForm.tsx (hunk 3 above)
Comment (Yadu-M) [reply src/components/ResumeForm.tsx]: No, i had the adding resume function nested inside the upload resume, i have it seperated now.

File: src/firebase/database.ts
Comment (Yadu-M) [reply src/firebase/database.ts:1]: Will fix

**Pull request merged**